src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -165,18 +176,7 @@ public bool TryComplete(SocketAsyncContext context)
             }
 
             public bool TrySetRunning()
-            {
-                State oldState = (State)Interlocked.CompareExchange(ref _state, (int)State.Running, (int)State.Waiting);
-                if (oldState == State.Cancelled)
-                {
-                    // This operation has already been cancelled, and had its completion processed.
-                    // Simply return false to indicate no further processing is needed.
-                    return false;
-                }
-
-                Debug.Assert(oldState == (int)State.Waiting);
```

Comment:
I wanted to make sure that we NEVER run into the situation where we set to running a non-waiting operation (Debug.Assert is not preventing this for Release builds)